rs/artifact_pool/src/certification_pool.rs

```diff
@@ -23,9 +23,71 @@ use ic_types::{
     consensus::HasHeight,
     Height,
 };
-use prometheus::IntCounter;
+use prometheus::{labels, opts, IntCounter, IntGauge};
 use std::collections::{BTreeMap, HashSet};
 
+struct PerTypeMetrics {
+    max_height: IntGauge,
+    min_height: IntGauge,
+    count: IntGauge,
+}
+
+const LABEL_TYPE: &str = "type";
+const LABEL_STAT: &str = "stat";
+
+impl PerTypeMetrics {
+    fn new(registry: &ic_metrics::MetricsRegistry, pool_portion: &str, type_name: &str) -> Self {
+        const NAME: &str = "artifact_pool_certification_height_stat";
+        const HELP: &str =
+            "The height of objects in a certification pool, by pool type, object type and stat";
+        Self {
+            max_height: registry.register(
+                IntGauge::with_opts(opts!(
+                    NAME,
+                    HELP,
+                    labels! {LABEL_POOL_TYPE => pool_portion, LABEL_TYPE => type_name, LABEL_STAT => "max"}
+                ))
+                .unwrap(),
+            ),
+            min_height: registry.register(
+                IntGauge::with_opts(opts!(
+                    NAME,
+                    HELP,
+                    labels! {LABEL_POOL_TYPE => pool_portion, LABEL_TYPE => type_name, LABEL_STAT => "min"}
+                ))
+                .unwrap(),
+            ),
+            count: registry.register(
+                IntGauge::with_opts(opts!(
+                    "certification_pool_size",
+                    "The number of artifacts in a certification pool, by pool type and object type",
+                    labels! {LABEL_POOL_TYPE => pool_portion, LABEL_TYPE => type_name}
+                ))
+                .unwrap(),
+            ),
+        }
+    }
+}
+
+/* FIXME: Remove PoolMetrics import and rename this */
+struct PoolMetrics {
+    certification: PerTypeMetrics,
+    certification_share: PerTypeMetrics,
+}
+
+impl PoolMetrics {
+    fn new(registry: ic_metrics::MetricsRegistry, pool_portion: &str) -> Self {
+        Self {
+            certification: PerTypeMetrics::new(&registry, pool_portion, "certification"),
+            certification_share: PerTypeMetrics::new(
+                &registry,
+                pool_portion,
+                "certification_share",
+            ),
+        }
+    }
```

Comment:
nit: we could have two methods here:
```
fn set_certifcaction_range(range: Range) {}
fn set_certification_shares_range(range: Range) {}
```
to remove some code duplication if `update_metrics` below

Reply:
Good idea, done